src/core/math/color.js

```diff
@@ -226,24 +226,31 @@ class Color {
      * is the same format as used in HTML/CSS.
      *
      * @param {boolean} alpha - If true, the output string will include the alpha value.
+     * @param {boolean} asArray - If true, the output will be an array of numbers.
      * @returns {string} The color in string form.
      * @example
      * const c = new pc.Color(1, 1, 1);
      * // Outputs #ffffffff
      * console.log(c.toString());
      */
-    toString(alpha) {
-        let s = `#${((1 << 24) + (Math.round(this.r * 255) << 16) + (Math.round(this.g * 255) << 8) + Math.round(this.b * 255)).toString(16).slice(1)}`;
+    toString(alpha, asArray) {
```

Comment:
Worth updating this?
https://github.com/playcanvas/engine/blob/main/test/core/math/color.test.mjs#L176